test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/DacFx/DacFxserviceTests.cs

```diff
@@ -537,6 +540,147 @@ public async void GenerateDeployScriptWithSqlCmdVariables()
+
+                VerifyAndCleanup(dacpacPath);
+            }
+            finally
+            {
+                sourceDb.Cleanup();
+                if (targetDb != null)
+                {
+                    targetDb.Cleanup();
+                }
+            }
+        }
+
+        private async Task VerifyDeployWithOptions(DeployParams deployParams, SqlTestDb targetDb, DacFxService service, ConnectionInfo connInfo, string expectedTableResult, string expectedViewResult)
+        {
+            var deployOperation = new DeployOperation(deployParams, connInfo);
+            service.PerformOperation(deployOperation, TaskExecutionMode.Execute);
+
+            using (SqlConnection conn = new SqlConnection(targetDb.ConnectionString))
+            {
+                try
+                {
+                    await conn.OpenAsync();
+                    var deployedResult = (string)ReliableConnectionHelper.ExecuteScalar(conn, $"SELECT TABLE_NAME FROM {targetDb.DatabaseName}.INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME = 'table3'; ");
+                    Assert.Equal(expectedTableResult, deployedResult);
+
+                    deployedResult = (string)ReliableConnectionHelper.ExecuteScalar(conn, $"SELECT TABLE_NAME FROM {targetDb.DatabaseName}.INFORMATION_SCHEMA.VIEWS WHERE TABLE_NAME = 'view1'; ");
+                    Assert.Equal(expectedViewResult, deployedResult);
+                }
+                finally
+                {
+                    conn.Close();
+                }
+            }
+        }
+
+        // <summary>
+        /// Verify that options are set correctly for a generate script request
+        /// </summary>
+        [Fact]
+        public async void GenerateDeployScriptWithOptions()
+        {
+            var result = GetLiveAutoCompleteTestObjects();
+            SqlTestDb sourceDb = await SqlTestDb.CreateNewAsync(TestServerType.OnPrem, query: SourceScript, dbNamePrefix: "DacFxDeployOptionsTestSource");
+            sourceDb.RunQuery(SourceViewScript);
+            SqlTestDb targetDb = await SqlTestDb.CreateNewAsync(TestServerType.OnPrem, query: TargetScript, dbNamePrefix: "DacFxDeployOptionsTestTarget");
+
+            try
+            {
+                DacFxService service = new DacFxService();
+                // First extract a db to have a dacpac to deploy later
+                string dacpacPath = InitialExtract(service, sourceDb, result);
+
+                // generate script to deploy the created dacpac with options
+                var generateScriptParams = new GenerateDeployScriptParams
+                {
+                    PackageFilePath = dacpacPath,
+                    DatabaseName = targetDb.DatabaseName,
+                    DeploymentOptions = new DeploymentOptions()
+                    {
+                        DropObjectsNotInSource = false,
```

Comment:
should we check once with true once with false so we know this really takes effect?